controlplane/kubeadm/controllers/controller.go

```diff
@@ -422,6 +435,43 @@ func (r *KubeadmControlPlaneReconciler) reconcileHealth(ctx context.Context, clu
 	return nil
 }
 
+// ReconcileInfrastructureMachineRefs adds infrastructure template ref as annotation to the infrastructure objects missing it.
+func (r *KubeadmControlPlaneReconciler) ReconcileInfrastructureMachineRefs(ctx context.Context, infraTemplateRef corev1.ObjectReference, machines internal.FilterableMachineCollection) (bool, error) {
+	isPatched := false
+	for _, machine := range machines {
+		if !machine.DeletionTimestamp.IsZero() || !machine.Status.InfrastructureReady {
+			continue
+		}
```

Comment:
Let's add a comment explaining why we're skipping over deleted and non-infrastructure ready machines.

Also cc @sethp-nr

Reply:
On second thought, even without having infrastructure ready, we can still annotate the machines. I see that check is not even done in reconcileHealth(). In which scenario, you think we need to wait for this condition?

Reply:
Makes sense, I'm not sure at this point why I thought that would have helped, let's remove it